### PR TITLE
Combined facets pr

### PR DESCRIPTION
**Pull request by deepunbxd005**

### CODE REVIEW

File: demo/src/config/searchConfigurations.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 const searchConfigurations = {
     siteName: 'wildearthclone-neto-com-au808941566310465',
     siteKey: 'e6959ae0b643d51b565dc3e01bf41ec1',
-    searchEndPoint: 'https://search.unbxd.io/',
+    searchEndPoint: 'https://console-nam.unbxd.io/ss/searchPreview/',
```
Comment (sanjithkumar017) [inline demo/src/config/searchConfigurations.js]: Please use 
`ab220086aca8111ea943906919cae33f-1069342559.us-west-1.elb.amazonaws.com82418d9335e92ca5f3208b8fbf58f71a/` as the end point.

File: src/modules/combinedFacets/CombinedFacetsContainer.js [hunk 2]
```diff
@@ -53,6 +77,133 @@ class CombinedFacetsContainer extends React.PureComponent {
...
         } = getRangeFacetCoreMethods(unbxdCore);
 
+        const {
+            getBucketedFacets,
+            getBreadCrumbsList,
+            setCategoryFilter,
+            deleteCategoryFilter,
+        } = getMultilevelFacetCoreMethods(unbxdCore);
+
+        // dfgerhgwerhgwerghwh
```
Comment (sanjithkumar017) [inline src/modules/combinedFacets/CombinedFacetsContainer.js]: why do we have this comment?

File: src/modules/combinedFacets/CombinedFacetsContainer.js [hunk 3]
```diff
@@ -53,6 +77,133 @@ class CombinedFacetsContainer extends React.PureComponent {
...
+                filterField,
+                position,
+                facetName,
+                facetType: facetTypes.MULTILEVEL_FACET,
+                values: [...breadCrumbFacets, ...formattedBucketedFacets]
+            };
+            multilevelFacets.push(facet);
+        });
+
+        //sgrwrherwherbnernre
```
Comment (sanjithkumar017) [inline src/modules/combinedFacets/CombinedFacetsContainer.js]: same here

File: demo/src/config/searchConfigurations.js (hunk 1 above)
Comment (deepunbxd005) [reply demo/src/config/searchConfigurations.js]: done

File: src/modules/combinedFacets/CombinedFacetsContainer.js (hunk 2 above)
Comment (deepunbxd005) [reply src/modules/combinedFacets/CombinedFacetsContainer.js]: removed

File: src/modules/combinedFacets/CombinedFacetsContainer.js (hunk 3 above)
Comment (deepunbxd005) [reply src/modules/combinedFacets/CombinedFacetsContainer.js]: removed

File: demo/src/components/CombinedFilters.js [hunk 4]
```diff
@@ -15,6 +17,8 @@ const CombinedFilters = () => {
             transform={transform}
             rangeFacetItemComponent={<RangeFacetItemComponent />}
             textFacetItemComponent={<TextFacetItemComponent />}
+            multiLevelFacetItemComponent={<MultiLevelItemComponent />}
+            multiLevelfacetClick={multiLevelfacetClick}
```
Comment (sanjithkumar017) [inline demo/src/components/CombinedFilters.js]: what about onClick for Text and Range facets?
Can we have a onFacetClick and handle clicks of 3 types of facets in that?

Comment (sanjithkumar017) [reply demo/src/components/CombinedFilters.js]: Add this to prop types.

Comment (sanjithkumar017) [reply demo/src/components/CombinedFilters.js]: How is `multiLevelfacetClick` being used?

File: src/modules/breadcrumbs/BreadcrumbsWrapper.js [hunk 5]
```diff
@@ -12,7 +12,6 @@ const BreadcrumbsWrapper = (props) => {
         separator,
         breadcrumbItemComponent
     } = props;
-
```
Comment (sanjithkumar017) [inline src/modules/breadcrumbs/BreadcrumbsWrapper.js:15]: Can you drop this file.

File: src/modules/combinedFacets/CombinedFacetsContainer.js [hunk 6]
```diff
@@ -15,6 +18,24 @@ import GenerateCombinedFacets from './generateCombinedFacets';
 import { executeCallback } from '../../common/utils';
 
 class CombinedFacetsContainer extends React.PureComponent {
+
+    componentDidMount(){
+        const {
+            categoryDisplayName = '',
```
Comment (sanjithkumar017) [inline src/modules/combinedFacets/CombinedFacetsContainer.js:24]: Are we will using these variables?

File: src/modules/combinedFacets/CombinedFacetsContainer.js [hunk 7]
```diff
@@ -15,6 +18,24 @@ import GenerateCombinedFacets from './generateCombinedFacets';
...
+            categoryField = '',
+            facetDepth,
+            facetLimit,
+            helpers: { setMultilevelFacetsConfiguration }
+        } = this.props;
+
+        setMultilevelFacetsConfiguration({
+            categoryDisplayName,
```
Comment (sanjithkumar017) [inline src/modules/combinedFacets/CombinedFacetsContainer.js:32]: Same here.

File: src/modules/combinedFacets/CombinedFacetsContainer.js [hunk 8]
```diff
@@ -32,7 +53,10 @@ class CombinedFacetsContainer extends React.PureComponent {
             enableViewMore,
             minViewMore,
             label,
-            onFacetClick
+            onFacetClick,
+            multiLevelfacetClick,
+            multiLevelFacetItemComponent,
```
Comment (sanjithkumar017) [inline src/modules/combinedFacets/CombinedFacetsContainer.js:57]: Add them to prop types

File: src/modules/combinedFacets/CombinedFacetsContainer.js [hunk 9]
```diff
@@ -70,7 +219,8 @@ class CombinedFacetsContainer extends React.PureComponent {
 
         const combinedFacets = [
             ...formattedTextFacets,
-            ...formattedRangeFacets
+            ...formattedRangeFacets,
+            ...multilevelFacets
```
Comment (sanjithkumar017) [inline src/modules/combinedFacets/CombinedFacetsContainer.js:222]: how are we utilising the new core sdk methods here? [ref](https://github.com/manu-unbxd/search-JS-core/blob/eda4de1bb72bb84f79b9c17bcabf1fc9f0950b49/src/modules/facets/getAllFacets.js)

File: src/modules/breadcrumbs/BreadcrumbsWrapper.js (hunk 5 above)
Comment (deepunbxd005) [reply src/modules/breadcrumbs/BreadcrumbsWrapper.js:15]: its being used in the container.

File: src/modules/combinedFacets/CombinedFacetsContainer.js (hunk 6 above)
Comment (deepunbxd005) [reply src/modules/combinedFacets/CombinedFacetsContainer.js:24]: its used in the method setMultilevelFacetsConfiguration

File: src/modules/combinedFacets/CombinedFacetsContainer.js (hunk 7 above)
Comment (deepunbxd005) [reply src/modules/combinedFacets/CombinedFacetsContainer.js:32]: its used in the setMultilevelFacetsConfiguration

File: src/modules/combinedFacets/CombinedFacetsContainer.js (hunk 8 above)
Comment (deepunbxd005) [reply src/modules/combinedFacets/CombinedFacetsContainer.js:57]: done

File: src/modules/combinedFacets/CombinedFacetsContainer.js (hunk 9 above)
Comment (deepunbxd005) [reply src/modules/combinedFacets/CombinedFacetsContainer.js:222]: discussed this issue offline.

File: demo/src/components/CombinedFilters.js (hunk 4 above)
Comment (deepunbxd005) [reply demo/src/components/CombinedFilters.js]: was not using the method, so removed it

File: demo/src/components/CombinedFilters.js [hunk 10]
```diff
@@ -3,7 +3,7 @@ import React from 'react';
 import { CombinedFacets } from '@unbxd-ui/react-search-sdk';
 import { FacetItemComponent as TextFacetItemComponent } from './TextFilters';
 import { FacetItemComponent as MultiLevelItemComponent } from './MultilevelFilters';
-import { onFacetClick as multiLevelfacetClick } from './MultilevelFilters';
+
```
Comment (sanjithkumar017) [inline demo/src/components/CombinedFilters.js:6]: Why is this removed?

File: src/modules/combinedFacets/CombinedFacetsContainer.js (hunk 6 above)
Comment (sanjithkumar017) [reply src/modules/combinedFacets/CombinedFacetsContainer.js:24]: But do we require it here?

File: src/modules/combinedFacets/CombinedFacetsContainer.js (hunk 7 above)
Comment (sanjithkumar017) [reply src/modules/combinedFacets/CombinedFacetsContainer.js:32]: But do we require it here?
